custom_components/netdaemon/climate.py

```diff
@@ -0,0 +1,211 @@
+"""Climate platform for NetDaemon."""
+from typing import TYPE_CHECKING
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_TEMPERATURE,
+)
+from homeassistant.const import ATTR_SUPPORTED_FEATURES
+
+from .const import (
+    ATTR_ATTRIBUTES,
+    ATTR_CLIENT,
+    ATTR_COORDINATOR,
+    ATTR_ENTITY_ID,
+    ATTR_STATE,
+    ATTR_TARGET_TEMPERATURE,
+    ATTR_TARGET_HUMIDITY,
+    DOMAIN,
+    TEMP_CELSIUS,
+    LOGGER,
+    PLATFORM_CLIMATE,
+)
+from .entity import NetDaemonEntity
+
+if TYPE_CHECKING:
+    from homeassistant.config_entries import ConfigEntry
+    from homeassistant.core import HomeAssistant
+    from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+    from .client import NetDaemonClient
+
+
+async def async_setup_entry(
+    hass: "HomeAssistant", _config_entry: "ConfigEntry", async_add_devices
+) -> None:
+    """Setup switch platform."""
+    client: "NetDaemonClient" = hass.data[DOMAIN][ATTR_CLIENT]
+    coordinator: "DataUpdateCoordinator" = hass.data[DOMAIN][ATTR_COORDINATOR]
+
+    climate_entities = []
+    for entity in client.entities:
+        if entity.split(".")[0] == PLATFORM_CLIMATE:
+            LOGGER.debug("Adding %s", entity)
+            climate_entities.append(
+                NetDaemonClimateEntity(coordinator, entity.split(".")[1])
+            )
+
+    if climate_entities:
+        async_add_devices(climate_entities)
+
+
+class NetDaemonClimateEntity(NetDaemonEntity, ClimateEntity):
+    """NetDaemon ClimateEntity class."""
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        if not self.entity_id:
+            return 0
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_SUPPORTED_FEATURES, 0)
+        )
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
```

Comment:
Should this be dynamic?
i.e be able to control with the service call?
```suggestion
        if not self.entity_id:
            return 0
        return (
            self._coordinator.data[self.entity_id]
            .get(ATTR_ATTRIBUTES, {})
            .get(ATTR_TEMPERATURE_UNIT, TEMP_CELSIUS)
        )
```
`ATTR_TEMPERATURE_UNIT` would be needed if suggestion is accepted

Reply:
Good suggestion!